server/src/main/java/org/apache/iotdb/db/utils/ErrorHandlingUtils.java

```diff
@@ -106,6 +107,9 @@ public static TSStatus tryCatchQueryException(Exception e) {
     } else if (e instanceof IoTDBException && !(e instanceof StorageGroupNotReadyException)) {
       DETAILED_FAILURE_QUERY_TRACE_LOGGER.warn(INFO_QUERY_PROCESS_ERROR, e);
       return RpcUtils.getStatus(((IoTDBException) e).getErrorCode(), getRootCause(e));
+    } else if (e instanceof TsFileRuntimeException) {
+      DETAILED_FAILURE_QUERY_TRACE_LOGGER.warn(INFO_QUERY_PROCESS_ERROR, e);
```

Comment:
If an aggregation result query throws an UnSupportedException, we should return a proper error code instead of 500.